scripts/spot-grid.py

```diff
@@ -0,0 +1,276 @@
+            spot_temp = numpy.copy(-spot)
+            ctx.arc(0, 0, 0.0025, 0, 2 * numpy.pi)
+            ctx.fill()
+        text_surface = cairo.ImageSurface(cairo.FORMAT_ARGB32, width, height)
+        info = [
+            "rotation: {:.1f} deg".format(numpy.rad2deg(self.rotation)),
+            "pitch-x: {:.2f} um".format(PIXEL_SIZE_UM * self.scaling[0]),
+            "pitch-y: {:.2f} um".format(PIXEL_SIZE_UM * self.scaling[1]),
+            "translation-x: {:.1f} um".format(PIXEL_SIZE_UM * self.translation[0]),
+            "translation-y: {:.1f} um".format(PIXEL_SIZE_UM * self.translation[1]),
+        ]
+        ctx2 = cairo.Context(text_surface)
+        ctx2.set_source_rgb(1.00, 0.83, 0.00)
+        font_size = 20
+        ctx2.set_font_size(font_size)
+        ctx2.translate(font_size, font_size)
+        # Cairo doesn't do multiline text plotting, so loop over the text and show at a lower location.
+        for text in info:
+            ctx2.translate(0, font_size)
+            ctx2.show_text(text)
+            ctx2.stroke()
+
+        if wx.MAJOR_VERSION <= 3:
+            self.imageCtrl.SetBitmap(wx.BitmapFromImage(self.img))
+            self.imageCtrlSpots.SetBitmap(wx.lib.wxcairo.BitmapFromImageSurface(spot_surface))
+            self.imageCtrlText.SetBitmap(wx.lib.wxcairo.BitmapFromImageSurface(text_surface))
+        else:
+            self.imageCtrl.SetBitmap(wx.Bitmap(self.img))
+            self.imageCtrlSpots.SetBitmap(wx.lib.wxcairo.BitmapFromImageSurface(spot_surface))
+            self.imageCtrlText.SetBitmap(wx.lib.wxcairo.BitmapFromImageSurface(text_surface))
+
+    def OnKey(self, event):
+        key = event.GetKeyCode()
+        if key in [ord("q"), ord("Q")]:
+            self.frame.Destroy()
+
+        # everything else we don't process
+        event.Skip()
+
+
+class ImagePasser(object):
+    def __init__(self):
+        self.image = None
+        self.available = threading.Event()
+        self.display = True
+
+
+def image_update(imp, window, gridsize=None):
+    try:
+        while imp.display:
+            imp.available.wait()
+            imp.available.clear()
+            if not imp.display:
+                return
+            window.new_image(imp.image, gridsize)
+    except Exception:
+        logging.exception("Failure during display")
+    finally:
+        logging.debug("Display thread ended")
+
+
+def live_display(ccd, dataflow_name, kill_ccd=True, gridsize=None):
+    """
+    Acquire an image from one (or more) dataflow and display it with a spot grid overlay.
+    ccd: a camera object
+    dataflow_name: name of the dataflow to access
+    kill_ccd: True if it is required to terminate the ccd after closing the window
+    gridsize: size of the grid of spots.
+    """
+    dataflow = getattr(ccd, dataflow_name)
+
+    size = ccd.resolution.value
+    # create a window
+    window = VideoDisplayer("Live from %s.%s" % (ccd.role, dataflow_name), size)
```

Comment:
Pass the `grid_size` to the VideoDisplayer, and then you can avoid passing it to the update_image thread, which passes it to the `new_image`.